### DISCUSSION TITLE
NullPointerException

### DISCUSSION THREAD

**Discussion by gw2366** [community/community#130578] [Programming Help] [answered]
### Body

I'm encountering a NullPointerException, and I can't figure out why. The exception is thrown in the following code snippet when I try to access an element in an array

```
public class Example {
    private String[] data;

    public Example(int size) {
        data = new String[size];
    }

    public void addData(int index, String value) {
        data[index] = value;
    }

    public String getData(int index) {
        return data[index].toUpperCase();
    }
}
```

Whenever I call getData, it throws a NullPointerException. What am I doing wrong?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by johnraivenolazo** [comment, marked as answer]
I guess it occurs because you're trying to call toUpperCase() on an element in the data array that might be null. If you haven't initialized an element at the specified index, it will be null by default. You need to check if the element is null before calling toUpperCase().

```
public class Example {
    private String[] data;

    public Example(int size) {
        data = new String[size];
    }

    public void addData(int index, String value) {
        data[index] = value;
    }

    public String getData(int index) {
        if (data[index] == null) {
            return null;
        }
        return data[index].toUpperCase();
    }
}
```
